docs/source/_templates/sg_launcher_links.html

```diff
@@ -0,0 +1,28 @@
+{#
+  Custom launcher links for Sphinx-Gallery example pages.
+  This adds an "Open in Colab" button that opens the notebook version
+  of the current example.
+#}
+
+{% if pagename.startswith('auto_examples/') and not pagename.endswith('/index') and pagename != 'auto_examples/index' %}
+
+{#
+  Derive the source .py path from the example page path.
+  Gallery pages live at auto_examples/<section>/<name> and source scripts at examples/<section>/<name>.py
+#}
+{% set example_py_path = pagename.replace('auto_examples/', 'examples/') + '.py' %}
+{% set example_ipynb_repo_path = example_py_path.replace('examples/', 'examples/_notebooks/').replace('.py', '.ipynb') %}
+
+<div class="sd-card sd-shadow-sm sd-mb-3">
+  <div class="sd-card-body">
+    <p class="sd-card-text sd-fs-5 sd-font-weight-bold sd-mb-2">Run this example</p>
+
+    {# Primary: open the source .py on Colab (Colab can render scripts; install instructions are in first cell) #}
+    <a class="sd-btn sd-btn-primary sd-mr-2" href="https://colab.research.google.com/github/deepinv/deepinv/blob/add_colab_badge/{{ example_ipynb_repo_path }}" target="_blank" rel="noopener">
+      <img alt="Open in Colab" src="https://colab.research.google.com/assets/colab-badge.svg" style="vertical-align: middle; height: 20px;"><br>
+      Try it!
```

Comment:
Maybe we could also add this link to the note at the top of the example, in conf?

Either in this sphinx option in conf:

https://github.com/deepinv/deepinv/blob/432f0d7ef7af4f3e36e46b7633661408c889b138/docs/source/conf.py#L344

Or by monkey patching the EXAMPLE string as I suggested in #657 .